tiatoolbox/dataloader/wsireader.py

```diff
@@ -121,65 +124,86 @@ def relative_level_scales(self, target_scale, units):
         """
         info = self.slide_info
 
+        def make_into_array(x):
+            """Ensure input x is a numpy array of length 2."""
+            if isinstance(x, numbers.Number):
+                # If one number is given, the same value is used for x and y
+                return np.array([x] * 2)
+            return np.array(x)
+
+        @np.vectorize
+        def level_to_downsample(x):
+            """Get the downsample for a level, interpolating non-integer levels."""
+            if isinstance(x, int) or int(x) == x:
+                # Return the downsample for the level
+                return info.level_downsamples[int(x)]
+            # Linearly interpolate between levels
+            floor = int(np.floor(x))
```

Comment:
Do we need to mention default interpolating behaviour is floor for rounding in the docs ?

Reply:
The floor here doesn't affect the interpolation method. It is just to find the boundary levels either side of resolution as the numpy method needs this. Eg for 1.5 the bounds are 1 and 2, the floor and the ceil of 1.5. The numpy method can then interpolate Newsweek the corresponding downsamples.